bot/feedback/feedback.py

```diff
@@ -0,0 +1,303 @@
+
+        Allows the owner to mark a suggestion as approved and add an optional reason to it. If this is the case, the
+        corresponding embed will be adapted to represent this change and the user who submitted the idea will be
+        notified via DM.
+
+        Args:
+            ctx (discord.ext.commands.Context): The context in which the command was called.
+            suggestion_id (int): The id of the suggestion which should be approved.
+            reason (Optional[str]): The reason for this decision.
+        """
+        await ctx.message.delete()
+        await self._change_suggestion_status(SuggestionStatus.APPROVED, ctx.author, ctx.guild, suggestion_id, reason)
+
+        log.info("Suggestion #%s has been approved by %s.", suggestion_id, ctx.author)
+
+    @manage_suggestions.command(name="deny")
+    @commands.is_owner()
+    @command_log
+    async def suggestion_deny(self, ctx: commands.Context, suggestion_id: int, *, reason: Optional[str]):
+        """Command Handler for the `suggestion` subcommand `deny`.
+
+        Allows the owner to mark a suggestion as denied and add an optional reason to it. If this is the case, the
+        corresponding embed will be adapted to represent this change and the user who submitted the idea will be
+        notified via DM.
+
+        Args:
+            ctx (discord.ext.commands.Context): The context in which the command was called.
+            suggestion_id (int): The id of the suggestion which should be approved.
+            reason (Optional[str]): The reason for the decision.
+        """
+        await ctx.message.delete()
+        await self._change_suggestion_status(SuggestionStatus.DENIED, ctx.author, ctx.guild, suggestion_id, reason)
+
+        log.info("Suggestion #%s has been denied by %s.", suggestion_id, ctx.author)
+
+    @manage_suggestions.command(name="consider")
+    @commands.is_owner()
+    @command_log
+    async def suggestion_consider(self, ctx: commands.Context, suggestion_id: int, *, reason: Optional[str]):
+        """Command Handler for the `suggestion` subcommand `consider`.
+
+        Allows the owner to mark a suggestion as considered and add an optional reason to it. If this is the case, the
+        corresponding embed will be adapted to represent this change and the user who submitted the idea will be
+        notified via DM.
+
+        Args:
+            ctx (discord.ext.commands.Context): The context in which the command was called.
+            suggestion_id (int): The id of the suggestion which should be approved.
+            reason (Optional[str]): The reason for the decision.
+        """
+        await ctx.message.delete()
+        await self._change_suggestion_status(SuggestionStatus.CONSIDERED, ctx.author, ctx.guild, suggestion_id, reason)
+
+        log.info("Suggestion #%s is being considered by %s.", suggestion_id, ctx.author)
+
+    @manage_suggestions.command(name="implemented")
```

Comment:
I am not sure if it is better to have 5 different methods for the status or to just have one and parse the status as an argument.
Maybe if you specify the type to SuggestionStatus in the parameter list and make sure it has the right values, it might even auto-parse.

Reply:
Doesn't work because the suggestion command has a keyword-only argument which would make it really hard to parse all of the arguments. I could make another subcommand like `set` or something like that, but then the command would only get longer. Example: `!suggestion set approve` instead of simply `!suggestion approve`